src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -45,23 +51,58 @@ export class QuerySummary extends Component {
    */
   static options: IQuerySummaryOptions = {
     /**
-     * Specifies whether to display the search tips to the end user when there are no search results.
+     * Specifies whether to hide the number of returned results.
+     *
+     * When this option is set to true, the number of returned results will be hidden from the page, meaning that your end users will not know how many results were returned for their query.
+     *
+     * Default value is `false`.
+     */
+    onlyDisplaySearchTips: ComponentOptions.buildBooleanOption({ defaultValue: false }),
+
+    /**
+     * Specifies whether to display the {@link QuerySummary.options.noResultsFoundMessage} message when there are no search results.
      *
      * Default value is `true`.
      */
-    enableSearchTips: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+    enableNoResultsFoundMessage: ComponentOptions.buildBooleanOption({ defaultValue: true }),
 
     /**
-     * Specifies whether to hide the information about the currently displayed range of results and only display the
-     * search tips instead.
+     * Specifies a custom message to display when there are no search results.
      *
-     * Default value is `false`.
+     * You can refer to the query the end user has entered using the `${query}` query tag.
+     *
+     * **Example**
+     * > For the `noResultFoundMessage` option, you enter `There were no results found for "${query}"`.
+     * > Your end user searches for `query without results`, which does not return any result.
+     * > On your page, they see this message: `There were no results found for "query without results"`.
+     *
+     * Default value is `No results for ${query}`.
+     */
+    noResultsFoundMessage: ComponentOptions.buildStringOption({
+      defaultValue: l('noResultFor', '${query}'),
+      depend: 'enableNoResultsFoundMessage'
+    }),
+
+    /**
+     * Specifies whether to display the `Cancel last action` link when there are no search results.
+     *
+     * When clicked, the link restores the previous query that contained results.
+     *
+     * Default value is `true`.
+     */
+    enableCancelLastAction: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+
+    /**
+     * Specifies whether to display search tips when there are no search results.
+     *
+     * Default value is `true`.
      */
-    onlyDisplaySearchTips: ComponentOptions.buildBooleanOption({ defaultValue: false })
+    enableSearchTips: ComponentOptions.buildBooleanOption({ defaultValue: true })
   };
 
   private textContainer: HTMLElement;
   private lastKnownGoodState: any;
+  private noResultsContainerSnapshot: string;
```

Comment:
What do you think about renaming `noResultsContainerSnapshot`  to `noResultsSnapshot` ?